spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -974,9 +1033,21 @@ struct ProductionVisitor
             pushState(std::move(pstate));
         }
 
+        if ( const auto& skipPre = anyProperty(p.unitType(), {"%skip-pre", "%skip"}) ) {
+            auto ctor = skipPre->expression()->as<hilti::expression::Ctor>().ctor();
+            if ( auto c = ctor.tryAs<hilti::ctor::RegExp>() )
+                skipRegExp(*c);
+        }
+
         for ( const auto& i : p.fields() )
             parseProduction(i);
 
+        if ( const auto& skipPost = anyProperty(p.unitType(), {"%skip-post", "%skip"}) ) {
```

Comment:
I think we have a misunderstanding about `%skip` here: it should't skip only at the beginning/end, but also in between fields. I suggest this logic:

```
if ( have %pre-skip )
  skip regex;

if ( have %skip )
  skip regex;

for ( const auto& i : p.fields() ) {
     parseProduction(i);

     if ( have %skip )
         skip regex;
}


if ( have %post-skip )
  skip regex;

```

\+ maybe some other way to resolve the case where we have both `%skip` *and*  `%pre/post-skip.`

Reply:
Another thing that's missing: disabling any of the %skip properties by assigning `Null`, see ticket and old tests.